app/controllers/providers/legal_aid_applications_controller.rb

```diff
@@ -5,8 +5,7 @@ class LegalAidApplicationsController < BaseController
 
     # GET /provider/applications
     def index
-      # TODO: update to return only current Provider's applications when provider can be identified
-      @applications = LegalAidApplication.limit(10)
+      @applications = LegalAidApplication.where(provider: current_provider).limit(10)
```

Comment:
I'd suggest:
```ruby
@applications = current_provider.legal_aid_applications.limit(10)
```

Reply:
I would also add a TODO to add pagination at some point

Reply:
Also, which 10 applications are fetched? Ideally, it should be the 10 most recent.
`current_provider.legal_aid_applications.order(created_at: :desc).limit(10)`
But maybe it is already the case by default?